ludwig/features/numerical_feature.py

```diff
@@ -207,11 +207,14 @@ def __init__(self, feature, encoder_obj=None):
     def forward(self, inputs):
         assert isinstance(inputs, torch.Tensor)
         assert inputs.dtype == torch.float32 or inputs.dtype == torch.float64
-        assert len(inputs.shape) == 1 or (len(inputs.shape) == 2 and inputs.shape[1] == 1)
+        assert len(inputs.shape) == 1 or (
+                    len(inputs.shape) == 2 and inputs.shape[1] == 1)
 
         if len(inputs.shape) == 1:
             inputs = inputs[:, None]
-        inputs_encoded = self.encoder_obj(inputs)
+
+        # ensure inputs are torch.float32 to avoid problems with use of nn.Linear
+        inputs_encoded = self.encoder_obj(inputs.type(torch.float32))
```

Comment:
wonder if it's better to silently cast or to remove float64 from the assert, which would for the user to do the cst outside. @tgaddair what do you think?